lua/entities/gmod_wire_hstracer/init.lua

```diff
@@ -0,0 +1,119 @@
+AddCSLuaFile( "shared.lua" )
+include('shared.lua')
+
+ENT.WireDebugName = "HighSpeed Tracer"
+
+--[[
+	0 = Clk / Trace
+
+	1 = StartPosition X
+	2 = StartPosition Y
+	3 = StartPosition Z
+
+	4 = EndPosition X
+	5 = EndPosition Y
+	6 = EndPosition Z
+
+	7 = Mask
+	8 = CollisionGroup
+
+	9 = Hit
+	10 = HitSky
+	11 = HitWorld
+	12 = Fraction
+
+	13 = HitPos X
+	14 = HitPos Y
+	15 = HitPos Z
+
+	16 = HitNormal X
+	17 = HitNormal Y
+	18 = HitNormal Z
+]]
+
+function ENT:Initialize()
+	self:PhysicsInit(SOLID_VPHYSICS)
+	self:SetMoveType(MOVETYPE_VPHYSICS)
+	self:SetSolid(SOLID_VPHYSICS)
+
+	self.tps = 0
+	self.tpsout = 0
+	self.tpstime = CurTime() + 1
+
+	self.Outputs = Wire_CreateOutputs(self, { "Memory" })
+	self.Memory = {}
+
+	self.Memory[7] = MASK_SOLID
+	self.Memory[8] = COLLISION_GROUP_NONE
+
+	self:UpdateOverlay()
+end
+
+function ENT:Trace()
+	self.BaseClass.Think(self)
```

Comment:
This shouldn't call baseclass think, the Think function should

Reply:
ah, I must've put it in there by accident thinking it was the Think method